jstz_cli/src/account/account.rs

```diff
@@ -7,31 +9,80 @@ use jstz_proto::context::account::Nonce;
 use serde::{Deserialize, Serialize};
 
 // Represents an individual account
+#[derive(Serialize, Deserialize, Debug, Clone, From)]
+pub enum Account {
+    Owned(OwnedAccount),
+    Alias(AliasAccount),
+}
+
 #[derive(Serialize, Deserialize, Debug, Clone)]
-pub struct Account {
+pub struct OwnedAccount {
     pub nonce: Nonce,
     pub alias: String,
     pub address: PublicKeyHash,
     pub secret_key: SecretKey,
     pub public_key: PublicKey,
 }
 
-impl Account {
-    pub fn from_passphrase(passphrase: String, alias: String) -> Result<Self> {
+impl OwnedAccount {
+    pub fn new(passphrase: String, alias: String) -> Result<Self> {
         let (sk, pk) = keypair_from_passphrase(passphrase.as_str()).unwrap();
 
-        println!("Secret key: {}", sk.to_string());
-        println!("Public key: {}", pk.to_string());
-
         let address = PublicKeyHash::try_from(&pk)?;
-        let new_account = Account {
+        let owned_account = Self {
             nonce: Nonce::default(),
             alias,
             address,
             secret_key: sk,
             public_key: pk,
         };
 
-        Ok(new_account)
+        Ok(owned_account)
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct AliasAccount {
+    pub alias: String,
+    pub address: PublicKeyHash,
+}
+
+impl AliasAccount {
+    pub fn new(address: String, name: String) -> Result<Self> {
+        let alias_account = Self {
+            alias: name,
+            address: PublicKeyHash::from_base58(address.as_str())?,
+        };
+
+        Ok(alias_account)
+    }
+}
+
+impl Account {
+    pub fn alias(&self) -> &str {
+        match self {
+            Account::Owned(OwnedAccount { alias, .. }) => alias,
+            Account::Alias(AliasAccount { alias, .. }) => alias,
+        }
+    }
+    pub fn address(&self) -> &PublicKeyHash {
+        match self {
+            Account::Owned(OwnedAccount { address, .. }) => address,
+            Account::Alias(AliasAccount { address, .. }) => address,
+        }
+    }
+
+    pub fn into_owned_mut(&mut self) -> Result<&mut OwnedAccount> {
+        match self {
+            Account::Owned(owned_account) => Ok(owned_account),
+            Account::Alias(_) => Err(anyhow!("Account is not owned")),
+        }
+    }
+
+    pub fn into_owned(&self) -> Result<&OwnedAccount> {
```

Comment:
Nit: Rust conventions (https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv). 

I think you want `as_owned` and `as_owned_mut`